extension/js/tv4.js

```diff
@@ -28,15 +27,80 @@ function tv4play_media_callback(data) {
   update_cmd();
 }
 
+
 matchers.push({
-  re: /^https?:\/\/(?:www\.)?tv4(?:play)?\.se\.?\/.*(?:-|\/)(\d+)/,
+  re: /^https?:\/\/(?:www\.)?tv4(?:play)?\.se\.?\/video\/([a-f0-9]+)/,
   func: function(ret) {
     const video_id = ret[1];
-    const data_url = `https://playback2.a2d.tv/asset/${video_id}?service=tv4&device=browser&browser=GoogleChrome&protocol=hls%2Cdash&drm=widevine&capabilities=live-drm-adstitch-2%2Cexpired_assets`;
+    console.log(video_id);
     update_filename(`${video_id}.mkv`);
-    update_json_url(data_url);
+    fetchMetaFile = function()
+    {
+        var tv4AccessToken = getWithExpiry(localStorageAccessTokenKey);
+        if (tv4AccessToken == null) {
+            error("Misslyckades h�mta en giltig tv4 access token");
+            return;
+        }
+        const metadataURL = `https://playback2.a2d.tv/play/${video_id}?service=tv4play&device=browser&protocol=hls%2Cdash&drm=widevine&browser=MicrosoftEdge&capabilities=live-drm-adstitch-2%2Cyospace3`;
+        fetch(metadataURL, {
+            method: 'GET',
+            headers: {
+                'X-Jwt': `Bearer ${tv4AccessToken}`
+            }
+        }).then(get_json).then(tv4play_media_callback).catch(api_error);
+    }
 
-    console.log(data_url);
-    fetch(data_url).then(get_json).then(tv4play_asset_callback).catch(api_error);
+    var tv4AccessTokenValue = getWithExpiry(localStorageAccessTokenKey);
+    if (tv4AccessTokenValue == null) {
+      var filter = { domain: "www.tv4play.se", name: "tv4-refresh-token" };
+      chrome.cookies.getAll(filter, function (cookies) {
```

Comment:
Istället för att hämta cookien med `chrome.cookies` så kan vi använda `chrome.tabs.executeScript`. Då slipper vi be om nya behörigheter.

```javascript
    chrome.tabs.executeScript({
      code: `document.cookie.split(';').map(c => c.trim()).find(c => c.startsWith("tv4-refresh-token="))?.split("=")[1]`
    }, function(refreshToken) {
      console.log('refreshToken', refreshToken);
    });
```